packages/core/src/tools/tool-builder/builder.test.ts

```diff
@@ -949,3 +949,49 @@ describe('Tool Input Validation', () => {
     expect(result.message).toContain('"tags": []');
   });
 });
+
+describe('CoreToolBuilder Output Schema', () => {
+  it('should allow ZodTuple in outputSchema', () => {
+    const mockModel = {
+      modelId: 'openai/gpt-4.1-mini',
+      provider: 'openrouter',
+      specificationVersion: 'v1',
+      supportsStructuredOutputs: false,
+    } as any;
+
+    const toolWithTupleOutput = createTool({
+      id: 'weather-tool',
+      description: 'Get weather information',
+      inputSchema: z.object({
+        location: z.string(),
+      }),
+      outputSchema: z.object({
+        temperature: z.number(),
+        conditions: z.string(),
+        test: z.tuple([z.string(), z.string()]),
+      }),
+      execute: async ({ context: _ }) => ({
```

Comment:
you can just remove variables if they're not being used
```suggestion
      execute: async () => ({
```